feat: Authenticate GutenbergKit WebView requests for private sites

dcalhoun opened the pull request:
## Description
<!-- Describe the changes, why they are needed, and how they address the issue -->
Provide GutenbergKit authentication cookies to allow loading protected resources
for private WordPress.com sites.

## Testing instructions
<!-- Consider listing specific testing steps for helping reviewers evaluate the efficacy of the changes -->
1. Open the editor for a private WordPress.com site.
1. Insert various media.
1. Verify the media displays/functions as expected.

<!--

Consider testing the following *yourself* before requesting a review:

- Compatibility with WordPress.com sites and self-hosted Jetpack sites
- Both portrait and landscape orientations
- Light and dark modes
- Dynamic font sizes: larger, smaller and bold text
- High contrast mode
- Screen reader experience (e.g., VoiceOver)
- Languages with large words or with letters/accents not frequently used in English
- Right-to-left languages layout
- Multi-tasking: split view and slide over

-->

dcalhoun committed:
feat: Authenticate WebView requests for private sites

Ensure WebView requests for private resources succeed.

dcalhoun committed:
test: Assert GutenbergKit authentication cookie configuration

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
build: Update GutenbergKit version

dcalhoun committed:
feat: Remove unnecessary host check

The private site feature is WPCOM-specific, so it implies the site is
hosted at WPCOM.
